### PR TITLE
Button: Deprecate 'description' prop

### LINKED ISSUE #4772: Standardize on "text", "secondaryText", and "tertiaryText"
A number of components have inconsistent naming conventions for rendering the text.

We have internally discussed and agreed we want to standardize all things:

Proposal:

Text should be rendered with the `text` prop.
Secondary text should be rendered with the `secondaryText` prop.

Work:

Persona: currently has "primaryText". This should be deprecated in favor of "text".
ContextualMenuItem: currently uses "name", deprecate in favor of "text"
Button: currently uses "description", deprecate in favor of "secondaryText"
ComboBox items: currently uses "value", deprecate in favor of "text"

These should be cleaned up, warnings added, and examples updated.

### PR DESCRIPTION
**Pull request by JasonGore**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #4772
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Deprecate Button's 'description' prop in favor of new 'secondaryText' prop. Replace all usages in codebase.

#### Focus areas to test

Button's use of description and secondaryText props.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx [hunk 1]
```diff
@@ -145,7 +147,7 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
     let ariaDescribedBy;
     if (ariaDescription) {
       ariaDescribedBy = _ariaDescriptionId;
-    } else if (description) {
+    } else if (secondaryText || description) {
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: you can also do this:

```
const {
  description,
  secondaryText = description
} = this.props;
```

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx [hunk 2]
```diff
@@ -313,11 +315,12 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
       text,
       children,
       description,
+      secondaryText,
       onRenderText = this._onRenderText,
       onRenderDescription = this._onRenderDescription
     } = this.props;
 
-    if (text || typeof (children) === 'string' || description) {
+    if (text || typeof (children) === 'string' || secondaryText || description) {
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: don't repeat yourself.

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx [hunk 3]
```diff
@@ -375,18 +378,19 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
...
     const {
-      description
+      description,
+      secondaryText
     } = props;
 
     // ms-Button-description is only shown when the button type is compound.
     // In other cases it will not be displayed.
-    return description ? (
+    return (secondaryText || description) ? (
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: It seems like we should have a component props filter in createComponent... doing this should be much more clean.

Comment (JasonGore) [reply packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: I thought so at first too, but the cases I've had to change have been relatively clean. As-is, removing the deprecated prop would just result in a few easy type errors to fix.

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx (hunk 1 above)
Comment (JasonGore) [reply packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: Thanks, I just first noticed this in another PR yesterday. Fixing.

**Pull request merged**